src/Hamiltonians/HubbardRealSpace.jl

```diff
@@ -181,17 +230,29 @@ function HubbardRealSpace(
         throw(ArgumentError("`u` must be a $C × $C matrix"))
     elseif size(t) ≠ (C,)
         throw(ArgumentError("`t` must be a vector of length $C"))
+    elseif length(v) ≠ C * D
+        throw(ArgumentError("`v` must be a $C × $D matrix"))
     elseif address isa BoseFS2C
         throw(ArgumentError(
             "`BoseFS2C` is not supported for this Hamiltonian, use `CompositeFS`"
         ))
     end
     warn_fermi_interaction(address, u)
 
-    u_mat = SMatrix{C,C,Float64}(u)
     t_vec = SVector{C,Float64}(t)
-    return HubbardRealSpace{C,typeof(address),typeof(geometry),typeof(t_vec),typeof(u_mat)}(
-        address, u_mat, t_vec, geometry,
+    u_mat = SMatrix{C,C,Float64}(u)
+    v_mat = SMatrix{C,D,Float64}(v)
+
+    # Precompute the trap potential terms
+    ranges = Tuple([range(-fld(M,2); length=M) for M in S],)
```

Comment:
```suggestion
    ranges = Tuple(range(-fld(M,2); length=M) for M in S)
```
This is equivalent and doesn't construct an array.